store.go

```diff
@@ -2051,6 +2060,13 @@ func dedupeNames(names []string) []string {
 }
 
 func (s *store) SetNames(id string, names []string) error {
+	opts := SetNameOptions{
+		AppendOnly: false,
+	}
```

Comment:
This is not a valid assumption, and could break untagging unless callers switch to `SetNamesWithOptions`.  An `AddNames/RemoveNames` pair instead of `SetNamesWithOptions()` would avoid this.

Reply:
implemented in below commit.